File: 1127-last-stone-weight/solution.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        heap = [-x for x in stones]
        heapq.heapify(heap)
        while len(heap) > 1:
            x = heapq.heappop(heap)
            y = heapq.heappop(heap)
            if x == y:
                continue
            else:
                val = -1 * abs(x - y)
                heapq.heappush(heap, val)
        if not heap:
            return 0
        else:
            return heap[0] * -1

File: 1127-last-stone-weight/solution.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        if (stones.empty()){
            return 0;
        }
        if (stones.size() == 1){
            return stones[0];
        }
        priority_queue<int> maxq;
        for(int stone : stones){
            maxq.push(stone);
        }
        int prev;
        while(maxq.size() > 1){
            prev = maxq.top();
            maxq.pop();
            prev = abs(prev - maxq.top());
            maxq.pop();
            if(prev > 0){
                maxq.push(prev);
            }
            
        }
        if(maxq.empty()){
            return 0;
        }
        return maxq.top();


    }
};
